src/redux/user/user.sagas.js

```diff
@@ -1,36 +1,172 @@
-import { call, put, takeEvery } from 'redux-saga/effects';
+import { call, put, takeEvery, delay } from 'redux-saga/effects';
 import { push } from 'connected-react-router';
-import { setUser, setUserError, setUserLoading } from './user.actions';
-import { LOGIN_USER } from './user.types';
+import {
+  setUser,
+  setUserError,
+  setUserLoading,
+  resetState,
+  setUserRecovered,
+  setPasswordReset,
+  setUserRegistered
+} from './user.actions';
+import {
+  LOGIN_USER,
+  CONFIRM_USER,
+  RECOVER_USER,
+  PASSWORD_RESET,
+  CHECK_IF_TOKEN_VALID,
+  REGISTER_USER
+} from './user.types';
 import { setItems } from '../../utils/client';
+import { SHOW_AFTER } from '../../configs/index';
 
-export const loginUser = (payload) => {
+export const loginUser = (data) => {
   const query = ` 
-  mutation login($user: UserInput!) {
+  mutation mutation($user: LoginInput!){
   loginUser(
-    user: $user
+    loginInput: $user
   ) {
     purchasedProducts
     orders
     token
-    id
+    _id
+    email
+    firstName
   }
 }
   `;
-  return setItems(query, payload);
+  return setItems(query, data);
+};
+
+export const confirmUser = (token) => {
+  const query = ` 
+  mutation confirmUser($token: String!){
+    confirmUser(token: $token)
+  }
+  `;
+  return setItems(query, { token });
+};
+
+export const recoverUser = (data) => {
+  const query = ` 
+  mutation recovery($email: String!, $language: Int!){
+    recoverUser(email: $email, language: $language)
+  }
+  `;
+  return setItems(query, data);
+};
+
+export const resetPassword = (data) => {
+  const query = ` 
+  mutation reset($password: String!, $token: String!){
+    resetPassword(password: $password, token: $token)
+  }
+  `;
+  return setItems(query, data);
+};
+
+export const checkIfTokenIsValid = (data) => {
```

Comment:
There is no sense to have it as a separate function because it is going to be used only once - inside the saga, please inline its code there